include/triton/Conversion/TritonGPUToLLVM/Utility.h

```diff
@@ -1010,7 +1010,30 @@ emitOffsetForWmmaLayout(const AMDWmmaEncodingAttr &wmmaLayout,
 }
 
 inline SmallVector<SmallVector<unsigned>>
-emitOffsetForLayout(Attribute layout, RankedTensorType type);
+emitOffsetForLayout(Attribute layout, RankedTensorType type) {
```

Comment:
Nit, could we keep this inside the .cc file?  This header is included in a lot of places, so when you change it you have to recompile a lot of Triton.

Reply:
I actually prefer to move many functions (in addition to this) to `.cc` files. Here I followed the old style to keep it inline.

Reply:
Please let me know if you prefer to change additional utility functions to `inline` or just this function?

Reply:
Sorry, I had a pending comment but forgot to click "send".

It's totally up to you and not a big deal.  The LL function had already been moved to the .cc file and the PR was  moving it back into the header, which seemed weird to me.  I don't think you need to clean up the whole header file though (unless you want to ofc).

Reply:
Done. Let's do big code refactoring later. I meant other functions like `emitCTAOffsetForLayout` are still `inline`, but maybe after linearlayout is fully integrated, these functions will be gone.